packages/vite/src/node/plugins/importAnalysisBuild.ts

```diff
@@ -535,74 +560,95 @@ export function buildImportAnalysisPlugin(config: ResolvedConfig): Plugin {
                     ? modulePreload === false
                       ? // CSS deps use the same mechanism as module preloads, so even if disabled,
                         // we still need to pass these deps to the preload helper in dynamic imports.
-                        [...deps].filter((d) => d.endsWith('.css'))
+                        [...deps].filter((d) =>
+                          getFileDep(d).url.endsWith('.css'),
+                        )
                       : [...deps]
                     : []
 
-                let renderedDeps: string[]
+                let renderedDeps: number[]
                 if (normalizedFile && customModulePreloadPaths) {
                   const { modulePreload } = config.build
                   const resolveDependencies = modulePreload
                     ? modulePreload.resolveDependencies
                     : undefined
-                  let resolvedDeps: string[]
+                  let resolvedDeps: number[]
                   if (resolveDependencies) {
                     // We can't let the user remove css deps as these aren't really preloads, they are just using
                     // the same mechanism as module preloads for this chunk
-                    const cssDeps: string[] = []
-                    const otherDeps: string[] = []
+                    const cssDeps: number[] = []
+                    const otherDeps: number[] = []
                     for (const dep of depsArray) {
-                      ;(dep.endsWith('.css') ? cssDeps : otherDeps).push(dep)
+                      ;(getFileDep(dep).url.endsWith('.css')
+                        ? cssDeps
+                        : otherDeps
+                      ).push(dep)
                     }
                     resolvedDeps = [
-                      ...resolveDependencies(normalizedFile, otherDeps, {
-                        hostId: file,
-                        hostType: 'js',
-                      }),
+                      ...resolveDependencies(
+                        normalizedFile,
+                        otherDeps.map((otherDep) => getFileDep(otherDep).url),
+                        {
+                          hostId: file,
+                          hostType: 'js',
+                        },
+                      ).map((otherDep) => addFileDep(otherDep)),
```

Comment:
Not a blocker. Question for @patak-dev: Looks like `resolveDependencies` is only implemented to return an array of strings, but it's [documented](https://vitejs.dev/config/build-options.html#build-modulepreload) as `(string | { runtime?: string })[]`. Is the documentation incorrect?

Reply:
Ah, yes, there was a big refactor in the lifetime of that PR and it kept the previous signature in the docs. Sent a PR to correct it https://github.com/vitejs/vite/pull/14562.